crates/bevy_tasks/src/iter/mod.rs

```diff
@@ -0,0 +1,444 @@
+use crate::TaskPool;
+
+mod adapters;
+pub use adapters::*;
+
+/// ParallelIterator closely emulates the std::iter::Iterator
+/// interface. However, it uses bevy_task to compute batches in parallel.
+pub trait ParallelIterator<B>
+where
+    B: Iterator<Item = Self::Item> + Send,
+    Self: Sized + Send,
+{
+    type Item;
+
+    fn next_batch(&mut self) -> Option<B>;
+
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        (0, None)
+    }
+
+    fn count(mut self, pool: &TaskPool) -> usize {
```

Comment:
I'm a little concerned people are going to end up putting extremely small units of work into some of these and blast it across threads and end up being a lot slower because of it. "Some people might misuse it" isn't a great reason to not do something but I feel kind of obligated to say it.

Reply:
Maybe I sgould add advice to the docs saying not to use ParallelIterator without profiling your code, because it has significant overhead relative to some tasks.

Reply:
Yeah a quick doc calling that out is a good idea. Its also worth adding a comment to the example, as many people learn using those.

Reply:
Ideally, I'd like the example to use much more than 128 sprites, since it's probably faster to just do the parallel_query example serially. However, for now, the limiting factor is how long it takes to draw a large number of sprites.

Reply:
I've added documentation in a few places calling out the overhead concerns with ParallelIterator.